txscript/sigvalidate.go

```diff
@@ -20,9 +20,14 @@ import (
 // pre-segwit, segwit v0, segwit v1 (taproot key spend validation), and the
 // base tapscript verification.
 type signatureVerifier interface {
-	// Verify returns true if the signature verifier context deems the
+	// Verify returns whether or not the signature verifier context deems the
 	// signature to be valid for the given context.
-	Verify() bool
+	Verify() verifyResult
```

Comment:
FWIW this only exists pre-segwit (doesn't apply to segwit v0 or v1). I wonder if we should implement this as a new super set interface? Downside is that it would end up bubbling up into the VM code (forced type assertion). 

Will give it a think over.